server/src/main/java/org/opensearch/indices/replication/OngoingSegmentReplications.java

```diff
@@ -147,6 +148,12 @@ void startSegmentCopy(GetSegmentFilesRequest request, ActionListener<GetSegmentF
      */
     CopyState prepareForReplication(CheckpointInfoRequest request, FileChunkWriter fileChunkWriter) throws IOException {
         final CopyState copyState = getCachedCopyState(request.getCheckpoint());
+        if (copyState.getCheckpoint().getCodec().equalsIgnoreCase(request.getCheckpoint().getCodec()) == false) {
```

Comment:
Why the case-insensitive comparison here?

Reply:
Good catch, that was a remnant of a previous version where we passed in a string of codec name instead of codec service. I'll make the change.